packages/fxa-react/lib/hooks.stories.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useCallback } from 'react';
+import { storiesOf } from '@storybook/react';
+import { useAwait, PromiseState } from './hooks';
+import LoadingSpinner from '../components/LoadingSpinner';
+
+const API_URL = '//worldtimeapi.org/api/ip';
+
+storiesOf('hooks/useAwait', module)
+  .add('basic', () => (
+    <>
+      <UseAwaitExample />
+      <UseAwaitExample apiUrl="//worldtimeapi.org/api/timezone/Europe/London" />
+      <UseAwaitExample apiUrl="//worldtimeapi.org/api/timezone/America/Detroit" />
+    </>
+  ))
+  .add('with initial state', () => (
+    <UseAwaitExample
+      initialState={{
+        result: initialResult(),
+        pending: false,
+        error: undefined,
+      }}
+    />
+  ))
+  .add('with immediate execution', () => (
+    <UseAwaitExample
+      executeImmediately={true}
```

Comment:
When actually being used I'd pass `executeImmediately` and not explicitly `={true}`, but that's just a syntax preference - I thought about suggesting not supplying a default so you have to pass `executeImmediately` or `executeImmediately={false}` to keep it to a required boolean and not optional/possibly `undefined` for a better visual of what's going on and slightly more restrictive typing, but maybe that's more of a chore since we probably don't want to typically execute immediately.

Reply:
Yeah, most of the uses I had in mind for this thing were in response to actions. So, the most common use would not be immediately executed. A few examples that *would* be immediate might be initial loading of plans & subscriptions on app start. The rest would be initiated by form submissions and such